dev/NumberBox/NumberBox.cpp

```diff
@@ -136,10 +136,14 @@ void NumberBox::OnApplyTemplate()
         }
     }
 
-    if (const auto headerPresenter = GetTemplateChildT<winrt::ContentPresenter>(c_numberBoxHeaderName, controlProtected))
+    if(const auto header = Header())
     {
-        // Set presenter to enable lightweight styling of the headers margin
-        m_headerPresenter.set(headerPresenter);
+        // We have the header specified, so lets render it!
```

Comment:
If `Header` is an empty string, the header presenter will be unnecessarily realized.

Reply:
I've adjusted that, thanks for the feedback!